lib/reverse_words.rb

```diff
@@ -1,25 +1,50 @@
 # A method to reverse each word in a sentence, in place.
+require 'pry'
 def string_reverse(my_string)
   return if my_string == nil || my_string.length < 1
-    i = 0 #first ele
-    j = my_string.length - 1 #last ele
-    while i < j # until position of first ele meets last, so there is the center element in between
-      temp = my_string[i] #stores frist ele in temp
-      my_string[i] = my_string[j] #sets first ele to the last ele's position
-      my_string[j] = temp #sets last element to first ele's position
-      i += 1 #increment
-      j -= 1
-    end
-    return my_string
+  i = 0 #first ele
+  j = my_string.length - 1 #last ele
+  while i < j # until position of first ele meets last, so there is the center element in between
+    temp = my_string[i] #stores frist ele in temp
+    my_string[i] = my_string[j] #sets first ele to the last ele's position
+    my_string[j] = temp #sets last element to first ele's position
+    i += 1 #increment
+    j -= 1
+  end
+  return my_string
 end
 
+# "   evol  "
+
 def reverse_words(my_words)
+  # puts my_words
   unless my_words == nil || my_words.empty?
-    my_words.
-    string_reverse(my)
+    last = 0
+    first = 0
+    until last == my_words.length - 1
+      while my_words[first] == " " && first != my_words.length - 1
+        first += 1
+      end
+      last = first
+      while my_words[last] != " " && last != my_words.length - 1
+        last += 1
+      end
+      if first == last || last == my_words.length - 1
+        my_words[first..last] = string_reverse(my_words[first..last])
```

Comment:
I didn't see any comments on time and space complexity. As we saw in the first few slides of https://drive.google.com/file/d/0B__DV26QHsH4bWJmS1A0QXBad1U/view?usp=sharing because of using the substring method e.g. my_words[first..last], your space complexity will be O(n).

Reply:
Can you think of a solution by updating your string_reverse method to take two additional parameters first and last along with the complete my_words and then only reversing the characters in my_words from index first to index last?

Reply:
Once you've given it some thought, take a look at the solution at https://github.com/Ada-C10/reverse_words/blob/solution/lib/reverse_words.rb